### PR TITLE
[14.0] connector_importer_product: create attr value if missing + increase test cov

### PR DESCRIPTION
**Pull request by simahawk**
See atomic commits for details.

ref: qoq-1496

### CODE REVIEW

File: connector_importer_product/components/product_product/record_handler.py [hunk 1]
```diff
@@ -184,8 +185,13 @@ def _update_template_attributes(self, odoo_record, values, orig_values):
         else:
             odoo_record.product_template_attribute_value_ids = valid_tpl_attr_values
 
+    def _find_attr(self, attr_column, orig_values):
+        """Find matching attribute."""
+        attr_xid = sanitize_external_id(attr_column)
+        return self.env.ref(attr_xid)
```
Comment (mmequignon) [inline connector_importer_product/components/product_product/record_handler.py:200]: Shouldn't we use raise_if_not_found here?

File: connector_importer_product/components/product_product/record_handler.py [hunk 2]
```diff
@@ -238,10 +247,44 @@ def _find_attr_value(self, attr, attr_column, orig_values):
...
         if not attr_value and "_value_" not in orig_val:
             # 3rd try w/ auto generated xid
-            value = slugify_one(orig_val).replace("-", "_")
-            xid = f"{attr_column}_value_{value}"
-            attr_value_external_id = sanitize_external_id(xid)
-            attr_value = self.env.ref(attr_value_external_id, False)
+            attr_value_xid = self._make_attribute_value_xid(attr_column, orig_val)
+            attr_value = self.env.ref(attr_value_xid, False)
+        if not attr_value and self.create_attribute_value_if_missing:
+            attr_value = self._create_missing_attribute_value(
```
Comment (mmequignon) [inline connector_importer_product/components/product_product/record_handler.py:253]: why would `_find_attr_value` create the value?
Shouldn't this be handled by a `_find_or_create_value` method?

File: connector_importer_product/components/product_product/record_handler.py [hunk 3]
```diff
@@ -203,7 +203,6 @@ def _find_attr(self, attr_column, orig_values):
     def _find_attr_value(self, attr, attr_column, orig_values):
```
Comment (mmequignon) [inline connector_importer_product/components/product_product/record_handler.py]: there's a lot of codecov warnings in this file.

**Review by mmequignon — APPROVED** [review]
LG, just a few not-blocking comments

File: connector_importer_product/components/product_product/record_handler.py (hunk 3 above)
Comment (mmequignon) [reply connector_importer_product/components/product_product/record_handler.py]: eh la madonna
image

Comment (simahawk) [reply connector_importer_product/components/product_product/record_handler.py]: is not due to this change. I'm adding test coverage in fact. I forgot the tests/__init__ update

File: connector_importer_product/components/product_product/record_handler.py (hunk 1 above)
Comment (simahawk) [reply connector_importer_product/components/product_product/record_handler.py:200]: no

File: connector_importer_product/components/product_product/record_handler.py (hunk 2 above)
Comment (simahawk) [reply connector_importer_product/components/product_product/record_handler.py:253]: fine

**Pull request merged**